### PR TITLE
Update web-animations IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2018-03-08 07:10:36
Finished: 2018-03-08 07:19:06


<h2>Failing Jobs</h2>
<ul>

<li>chrome:dev</li>

<li>safari:11.0</li>

<li>MicrosoftEdge:16.16299</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25544)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/350514405)

**Comment by lukebjerring** [comment]
Relies on https://github.com/w3c/web-platform-tests/pull/10100 for the `only` filtering of untested IDL.

**Comment by birtles** [comment]
Sorry, I'm missing some context. What is the purpose of this PR?

**Comment by lukebjerring** [comment]
Hi, sorry for the lack of context - we (@foolip, @mdittmer and I) are working on a pipeline which automatically scrapes known WebIDL sources/specs and updates the IDL in the interfaces whenever it changes.

This PR is the output of running the scraper as a one-off, for which I've included updating the existing/equivalent IDL snippets that are inlined in the tests, to instead reference the generated (singular) interfaces/web-animations.idl file.

Later, we'll use OWNERS to automatically ping relevant persons when the source changed and the scraped IDL differs from what's in WPT.

File: interfaces/web-animations.idl [hunk 1]
```diff
@@ -0,0 +1,177 @@
+// GENERATED CONTENT - DO NOT EDIT
+// Content of this file was automatically extracted from the Web Animations spec.
+// See https://w3c.github.io/web-animations/
```
Comment (birtles) [inline interfaces/web-animations.idl]: This should be https://drafts.csswg.org/web-animations/

File: interfaces/web-animations.idl [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    void play ();
+    void pause ();
+    void updatePlaybackRate (double playbackRate);
+    void reverse ();
+};
+
+enum AnimationPlayState { "idle", "running", "paused", "finished" };
+
+[Exposed=Window]
+interface AnimationEffectReadOnly {
```
Comment (birtles) [inline interfaces/web-animations.idl]: This interface no longer exists in the latest version of Web Animations.

File: interfaces/web-animations.idl [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    readonly attribute AnimationEffectTimingReadOnly timing;
+    ComputedTimingProperties getComputedTiming();
+};
+
+[Exposed=Window]
+interface AnimationEffectTimingReadOnly {
```
Comment (birtles) [inline interfaces/web-animations.idl]: Nor does this interface.

File: interfaces/web-animations.idl [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    readonly attribute FillMode                           fill;
+    readonly attribute double                             iterationStart;
+    readonly attribute unrestricted double                iterations;
+    readonly attribute (unrestricted double or DOMString) duration;
+    readonly attribute PlaybackDirection                  direction;
+    readonly attribute DOMString                          easing;
+};
+
+[Exposed=Window]
+interface AnimationEffectTiming : AnimationEffectTimingReadOnly {
```
Comment (birtles) [inline interfaces/web-animations.idl]: Nor this one.

**Review by birtles — CHANGES_REQUESTED** [review]
Thanks for doing this but I think the scraper needs to be run again. It appears to be using an old version of the spec.

**Comment by birtles** [comment]
Also, one question about the `GENERATED CONTENT - DO NOT EDIT` part:

Often I'll make spec, test, and code changes simultaneously, update the tests/code in Gecko, then upstream the tests to wpt. Can I rerun the scraper locally so that I can land the updated tests along with the code in Gecko?

**Comment by lukebjerring** [comment]
> Also, one question about the GENERATED CONTENT - DO NOT EDIT part:

NB: If you copy-paste from the source, strip leading whitespace, single blank line between snippets, your output will exactly match anyway. The comment is targeted at those who are unfamiliar with the spec, so as not to move the file.

> Often I'll make spec, test, and code changes simultaneously, update the tests/code in Gecko, then upstream the tests to wpt. Can I rerun the scraper locally so that I can land the updated tests along with the code in Gecko?

Desired / future answer: Yes
Current answer: No

When the tool matures (it still has a few edge-cases with whitespace, URLs, etc) we'll standardize how it gets run, document it, and ensure that changes can be included with test updates.

File: interfaces/web-animations.idl (hunk 1 above)
Comment (lukebjerring) [reply interfaces/web-animations.idl]: I'll defer that change for @foolip

**Comment by birtles** [comment]
Looks good. I'll wait for the change to the header before marking it as approved.

Regarding editing the files, as long as I can make (small, careful) hand edits, that sounds fine.

File: interfaces/web-animations.idl (hunk 1 above)
Comment (foolip) [reply interfaces/web-animations.idl]: https://w3c.github.io/web-animations/ now redirects to https://drafts.csswg.org/web-animations/ but not using a HTTP redirect.

Sent two reviews to update URLs I found in Chromium, only the first actually relevant:
https://chromium-review.googlesource.com/c/chromium/src/+/970648
https://github.com/web-animations/web-animations-js/pull/191

@lukebjerring, have the other recent changes taken effect in the output of the IDL scraper? If so, then after the first has landed, another update should fix this.

Comment (foolip) [reply interfaces/web-animations.idl]: Ah, except the day-to-day has the wrong URL as well. Will fix.

Comment (lukebjerring) [reply interfaces/web-animations.idl]: Updated the source (switched https://drafts.csswg.org/web-animations/ to be the canonical source and https://w3c.github.io/web-animations/ to be an `alt` source)

**Comment by lukebjerring** [comment]
@birtles - updated header. Thanks for taking the time to review!

**Review by foolip — APPROVED** [review]

**Review by birtles — APPROVED** [review]
Thanks!

**Pull request merged**